docs/csharp/fundamentals/functional/deconstruct.md

```diff
@@ -0,0 +1,102 @@
+
+:::code language="csharp" source="./snippets/deconstructing-tuples/deconstruct-tuple2.cs" ID="Snippet1":::
+
+- You can use the `var` keyword so that C# infers the type of each variable. You place the `var` keyword outside of the parentheses. The following example uses type inference when deconstructing the three-tuple returned by the `QueryCityData` method.
+
+    :::code language="csharp" source="./snippets/deconstructing-tuples/deconstruct-tuple3.cs" ID="Snippet1":::
+
+    You can also use the `var` keyword individually with any or all of the variable declarations inside the parentheses.
+
+    :::code language="csharp" source="./snippets/deconstructing-tuples/deconstruct-tuple4.cs" ID="Snippet1":::
+
+    This is cumbersome and isn't recommended.
+
+- Lastly, you may deconstruct the tuple into variables that have already been declared.
+
+    :::code language="csharp" source="./snippets/deconstructing-tuples/deconstruct-tuple5.cs" ID="Snippet1":::
+
+You can't specify a specific type outside the parentheses even if every field in the tuple has the
+same type. Doing so generates compiler error CS8136, "Deconstruction 'var (...)' form disallows a specific type for 'var'.".
+
+You must assign each element of the tuple to a variable. If you omit any elements, the compiler generates error CS8132, "Can't deconstruct a tuple of 'x' elements into 'y' variables."
+
+You can't mix declarations and assignments to existing variables on the left-hand side of a deconstruction. The compiler generates error CS8184, "a deconstruction can't mix declarations and expressions on the left-hand-side." when the members include newly declared and existing variables.
+
+## Tuple elements with discards
+
+Often when deconstructing a tuple, you're interested in the values of only some elements. Starting with C# 7.0, you can take advantage of C#'s support for *discards*, which are write-only variables whose values you've chosen to ignore. A discard is chosen by an underscore character ("\_") in an assignment. You can discard as many values as you like; all are represented by the single discard, `_`.
+
+The following example illustrates the use of tuples with discards. The `QueryCityDataForYears` method returns a six-tuple with the name of a city, its area, a year, the city's population for that year, a second year, and the city's population for that second year. The example shows the change in population between those two years. Of the data available from the tuple, we're unconcerned with the city area, and we know the city name and the two dates at design-time. As a result, we're only interested in the two population values stored in the tuple, and can handle its remaining values as discards.  
+
+:::code language="csharp" source="./snippets/deconstructing-tuples/deconstruct-tuple1.cs":::
```

Comment:
The snippet doesn't match the text introducing it; text speaks of `QueryCityDataForYears` that returns a 6-tuple, snippet omits ForYears and returns a 3-tuple.